flusher.go

```diff
@@ -45,7 +51,11 @@ func (s *Server) FlushGlobal(interval time.Duration, metricLimit int) {
 
 	s.reportGlobalMetricsFlushCounts(ms)
 
-	go s.flushS3(finalMetrics)
+	go func() {
+		for _, p := range s.getPlugins() {
+			go p.Flush(finalMetrics, s.Hostname)
```

Comment:
Yup!

Also, even without a mutex, this means that, no matter how many plugins we have loaded, the main `Flush` goroutine only creates one goroutine per flush, which (may) have a slight performance improvement over creating O(n) additional goroutines in the main goroutine per flush.